tests/elster_xml/grundsteuer/test_elster_data_representation.py

```diff
@@ -0,0 +1,335 @@
+from erica.request_processing.erica_input.v2.grundsteuer_input import Anrede, Anteil, Vertreter, Person, Eigentuemer
+from tests.sample_data import get_sample_vertreter_dict, get_single_person_dict, create_grundsteuer
+
+
+class TestElsterifyAnrede:
+    def test_no_anrede_is_correctly_translated(self):
+        result = elsterify_anrede(Anrede.no_anrede)
+        assert result == '01'
+
+    def test_herr_is_correctly_translated(self):
+        result = elsterify_anrede(Anrede.herr)
+        assert result == '02'
+
+    def test_frau_is_correctly_translated(self):
+        result = elsterify_anrede(Anrede.frau)
+        assert result == '03'
+
+    def test_invalid_value_raises_key_error(self):
+        with pytest.raises(KeyError):
+            elsterify_anrede("INVALID")
+
+
+class TestEAnteil:
+    def test_attributes_set_correctly(self):
+        anteil_obj = Anteil.parse_obj({'zaehler': '1', 'nenner': '2'})
+
+        result = EAnteil(anteil_obj)
+
+        assert result.E7404570 == anteil_obj.zaehler
+        assert result.E7404571 == anteil_obj.nenner
+
+
+class TestEGesetzlicherVertreter:
+    def test_attributes_set_correctly(self):
+        full_vertreter_obj = Vertreter.parse_obj(get_sample_vertreter_dict())
+
+        result = EGesetzlicherVertreter(full_vertreter_obj)
+
+        assert result.E7415101 == elsterify_anrede(full_vertreter_obj.name.anrede)
+        assert result.E7415102 == full_vertreter_obj.name.titel
+        assert result.E7415201 == full_vertreter_obj.name.vorname
+        assert result.E7415301 == full_vertreter_obj.name.name
+        assert result.E7415401 == full_vertreter_obj.adresse.strasse
+        assert result.E7415501 == full_vertreter_obj.adresse.hausnummer
+        assert result.E7415502 == full_vertreter_obj.adresse.hausnummerzusatz
+        assert result.E7415601 == full_vertreter_obj.adresse.plz
+        assert result.E7415602 == full_vertreter_obj.adresse.postfach
+        assert result.E7415603 == full_vertreter_obj.adresse.ort
+        assert result.E7415604 == full_vertreter_obj.telefonnummer.telefonnummer
+        assert len(vars(result)) == 11
+
+    def test_if_all_optional_attributes_not_given_then_attributes_set_correctly(self):
+        vertreter_obj = Vertreter.parse_obj(get_sample_vertreter_dict(complete=False))
+
+        result = EGesetzlicherVertreter(vertreter_obj)
+
+        assert result.E7415101 == elsterify_anrede(vertreter_obj.name.anrede)
+        assert result.E7415102 is None
+        assert result.E7415201 == vertreter_obj.name.vorname
+        assert result.E7415301 == vertreter_obj.name.name
+        assert result.E7415401 is None
+        assert result.E7415501 is None
+        assert result.E7415502 is None
+        assert result.E7415601 == vertreter_obj.adresse.plz
+        assert result.E7415602 is None
+        assert result.E7415603 == vertreter_obj.adresse.ort
+        assert result.E7415604 is None
+        assert len(vars(result)) <= 11
```

Comment:
OK, I understand. I'm still skeptical about the usefulness of this assertion as long as we don't have a deterministic expected value. What if an undesired var is added in the process, replacing the missing one? The assertion will still pass. In a way it could be even harmful: the test is a sort of specification with which we're indicating that we're confident that any value equal to or less than 11 is fine, whereas in reality that's not the case.

How about we integrate the asdict step (where CustomParser is used) with this step and run these tests after that step is executed?

Reply:
I found the error and the solution^^ In fact, it really was because of the non-existing parent telefonnummer object. I now handle optionals differently and thus can deterministically say how many attributes are set.